performance-test/src/puppeteer-test.ts

```diff
@@ -0,0 +1,363 @@
+  // we run it twice without measurements to warm up the environment
+  const [button] = await page.$x("//a[contains(., 'P S')]")
+  await button!.click()
+  await consoleDoneMessage(page)
+  const [button2] = await page.$x("//a[contains(., 'P S')]")
+  await button2!.click()
+  await consoleDoneMessage(page)
+  // and then we run the test for a third time, this time running tracing
+  await page.tracing.start({ path: 'trace.json' })
+  const [button3] = await page.$x("//a[contains(., 'P S')]")
+  await button3!.click()
+  await consoleDoneMessage(page)
+  await page.tracing.stop()
+  await browser.close()
+  let traceData = fs.readFileSync('trace.json').toString()
+  const traceJson = JSON.parse(traceData)
+
+  const frameTimeEvents: any[] = traceJson.traceEvents.filter((e: any) =>
+    e.name.startsWith('scroll_step_'),
+  )
+  let frameTimes: Array<number> = []
+  let lastFrameTimestamp: number | null = null
+  let totalFrameTimes = 0
+  frameTimeEvents.forEach((fte) => {
+    const frameID = fte.name.split('scroll_step_')[1] - 1
+    const frameTimestamp = fte.ts
+    if (lastFrameTimestamp != null) {
+      const frameDelta = (frameTimestamp - lastFrameTimestamp) / 1000
+      frameTimes[frameID] = frameDelta
+      totalFrameTimes += frameDelta
+    }
+    lastFrameTimestamp = frameTimestamp
+  })
+
+  let frameTimesarray = frameTimes.map((x) => Number(x.toFixed(1)))
+
+  const frameData = {
+    frameAvg: totalFrameTimes / frameTimesarray.length,
+    percentile25: frameTimesarray.sort((a, b) => a - b)[Math.floor(frameTimesarray.length * 0.25)],
+    percentile50: frameTimesarray.sort((a, b) => a - b)[Math.floor(frameTimesarray.length * 0.5)],
+    percentile75: frameTimesarray.sort((a, b) => a - b)[Math.floor(frameTimeEvents.length * 0.75)],
+  }
+
+  const imageFileName = v4() + '.png'
+
+  const fileURI = await createTestPng(frameTimesarray, imageFileName, frameData)
+  const s3FileUrl = await uploadPNGtoAWS(fileURI)
+
+  console.info(
+    `::set-output name=perf-result:: "![TestFrameChart](${s3FileUrl}) ${totalFrameTimes}ms – average frame length: ${
+      frameData.frameAvg
+    }
+      – Q1: ${frameData.percentile25} – Q2: ${frameData.percentile50} – Q3: ${
+      frameData.percentile75
+    } – Median: ${frameData.percentile50} – frame times: [${frameTimes
+      .sort((a, b) => a - b)
+      .join(',')}]"`,
+  )
+}
+
+function valueOutsideCutoff(frameCutoff: Array<number>) {
+  let sum = 0
+  for (let i = 0; i < frameCutoff.length; i++) {
+    if (frameCutoff[i]! > 130) {
+      sum += 1
+    }
+  }
+  return sum
+}
+
+async function createTestPng(
+  frameTimesArray: Array<number>,
+  testFileName: string,
+  frameData: {
+    frameAvg: number
+    percentile25: number | undefined
+    percentile50: number | undefined
+    percentile75: number | undefined
+  },
+) {
+  const plotly = require('plotly')('OmarDaSilva', 'hE7jUM0tDlgyByK5fdwH')
```

Comment:
is this an account and key? can you please move these to environment variables too?

Reply:
I haven't got access to secret key repo, I'll ask someone to do it for me